docs/continuous-integration/devcontainers/README.md

```diff
@@ -0,0 +1,64 @@
+| No surprise for the developers, local outputs (of linting for instance) will be the same in the CI | Some tools were installed via a definition in `devcontainer.json`, resulting in a different container from the dev environment |
+| All tooling and their versions defined in a single place                                           | Some pipeline tasks will not be available                                                                                      |
+| Tools/dependencies are already present                                                             | Building the image for each pipeline run is slow&#178;                                                                         |
+| The dev container is being tested to include all new tooling in addition to not being broken       |                                                                                                                                |
+
+> &#185;: container size can be reduces by exporting the layer that contains only the tooling needed for the CI pipeline
+> &#178;: could be mitigated via adding image caching without using a container registry
+
+### Run CI pipelines in the dev container with image registry
+
+| Pros                                                                                               | Cons                                                                                                                                 |
+| -------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| Utilities scripts will work out of the box                                                         | Need to rebuild the container for each run, given that there may be changes within the branch being built                            |
+| No surprise for the developers, local outputs (of linting for instance) will be the same in the CI | Not everything in the container is needed for the CI pipeline&#185;                                                                  |
+| Rules used (for linting or unit tests) will be the same on the CI                                  | Some tools were installed via a definition in `devcontainer.json`, resulting in a different container from the dev environment&#178; |
+| All tooling and their versions defined in a single place                                           | Some pipeline tasks will not be available &#179;                                                                                     |
+| Tools/dependencies are already present                                                             | Require access to a container registry to host the container within the pipeline&#8308;                                              |
```

Comment:
Is it worth noting that GitHub provides GHCR which can be used without setting up a separate registry?

Reply:
It is a mitigation of having to pass in your own registry credentials in order to use this in a pipeline. With Github Actions you can just use the built in, which is cool if you don't know that you can do that. 

Why do you think it's not worth it?

Reply:
Sorry, my "is it worth..." comment was really me suggesting it and asking whether you agreed that it would be good to add 😄

Reply:
I modified the notation on this below to be more detailed :)